### PR TITLE
iterate over PANIC design and ergonomics

### PR DESCRIPTION
**Pull request by siscia**
Follow up from #2424 

I believe is wiser to accept also the mask, this will allow us to use the PANIC more widely in the codebase.

### CODE REVIEW

**Comment by siscia** [comment]
Maybe we should also accept the paramenter for the LogSource? (this one: https://github.com/cvmfs/cvmfs/blob/devel/cvmfs/logging_internal.h#L71)

File: test/quickcheck/qc_catalog_merge_tool.cc [hunk 1]
```diff
@@ -104,8 +105,7 @@ DirSpec ModifySpec(const DirSpec& in) {
         // TODO(radu): Implement file content modifications
         break;
       default:
-        LogCvmfs(kLogCvmfs, kLogStderr, "Unknown change type. Aborting.");
-        abort();
+        PANIC("Unknown change type. Aborting.");
```
Comment (jblomer) [inline test/quickcheck/qc_catalog_merge_tool.cc]: Mask is missing

File: cvmfs/catalog_diff_tool_impl.h [hunk 2]
```diff
@@ -112,18 +113,16 @@ void CatalogDiffTool<RoCatalogMgr>::DiffRec(const PathString& path) {
     catalog::DirectoryEntry new_entry = new_listing[i_to];
 
     if (old_entry.linkcount() == 0) {
-      LogCvmfs(kLogCvmfs, kLogStderr,
-                "CatalogDiffTool - Entry %s in old catalog has linkcount 0. "
-                "Aborting.",
-                old_entry.name().c_str());
-      abort();
+      PANIC(kLogStderr,
+          "CatalogDiffTool - Entry %s in old catalog has linkcount 0. "
```
Comment (jblomer) [inline cvmfs/catalog_diff_tool_impl.h]: Indentation is (I think) accidentally off by one

File: cvmfs/catalog_diff_tool_impl.h [hunk 3]
```diff
@@ -112,18 +113,16 @@ void CatalogDiffTool<RoCatalogMgr>::DiffRec(const PathString& path) {
...
+          old_entry.name().c_str());
     }
     if (new_entry.linkcount() == 0) {
-      LogCvmfs(kLogCvmfs, kLogStderr,
-                "CatalogDiffTool - Entry %s in new catalog has linkcount 0. "
-                "Aborting.",
-                new_entry.name().c_str());
-      abort();
+      PANIC(kLogStderr,
+          "CatalogDiffTool - Entry %s in new catalog has linkcount 0. "
```
Comment (jblomer) [inline cvmfs/catalog_diff_tool_impl.h]: Indentation

**Comment by siscia** [comment]
Overall it seems smarter to allow the LogSource as well.

So we keep the interface the same and we can wide the scope of PANIC.

**Comment by siscia** [comment]
@jblomer do we like this final interface?

This will change in several places and it will be quite a bit of work, so let's make sure that we agree on this one :)

**Comment by jblomer** [comment]
I'm not sure about the log source.  The log source actually only makes sense for the debug log, when you try to parse a long log and quickly want to grep for things or skip others.  For `PANIC`, things come to a full stop, so we simply can use `kLogCvmfs` as the default source and not bother repeating it for every `PANIC` call.

**Comment by siscia** [comment]
I added because there are other places that use the same pattern: Log+Abort but they have a different source.

One example is the cache: https://github.com/cvmfs/cvmfs/blob/devel/cvmfs/cache_transport.cc#L510

But I believe I found others...

**Comment by jblomer** [comment]
Yep, `PANIC` and `LogCvmfs` would not be interchangeable function calls.  But in this case, I think the bigger benefit is not adding a log source to every `PANIC` call that, at the end of the day, is not needed.

**Comment by siscia** [comment]
I removed the source from the PANIC call, but I left it in the underline code.
This will allow us the flexibility to add other PANIC call if necessary tomorrow.

**Pull request merged**